templates/policy-compliance-agent/backend/services/embedding_manager.py

```
@@ -0,0 +1,202 @@
+        Returns:
+            Embedding vector as a list of floats
+        """
+        try:
+            if not text or not text.strip():
+                raise ValueError("Empty text provided to embed_single")
+
+            if not self.embedding_client:
+                logger.error(
+                    "Embedding client not available, cannot generate embedding"
+                )
+                raise ValueError("Embedding client not available")
```

Ruff (0.14.5)

Abstract `raise` to an inner function

(TRY301)